src/groundlight/client.py

```diff
@@ -206,26 +226,25 @@ def wait_for_confident_result(
         :param timeout_sec: The maximum number of seconds to wait.
         """
         # TODO: Add support for ImageQuery id instead of object.
-        timeout_time = time.time() + timeout_sec
-        delay = self.POLLING_INITIAL_DELAY
-        while time.time() < timeout_time:
-            current_confidence = image_query.result.confidence
-            if current_confidence is None:
-                logging.debug("Image query with None confidence implies human label (for now)")
+        start_time = time.time() 
+        next_delay = self.POLLING_INITIAL_DELAY
+        target_delay = 0
+        image_query = self._post_process_image_query(image_query)
+        while True:
```

Comment:
I think there was a logic bug in the delay calculations in here which we should extract.